aspnetcore/web-api/action-return-types.md

```diff
@@ -36,60 +36,97 @@ This document explains when it's most appropriate to use each return type.
 
 The simplest action returns a primitive or complex data type (for example, `string` or a custom object type). Consider the following action, which returns a collection of custom `Product` objects:
 
-[!code-csharp[](../web-api/action-return-types/samples/WebApiSample.Api.21/Controllers/ProductsController.cs?name=snippet_Get)]
+[!code-csharp[](../web-api/action-return-types/samples/2x/WebApiSample.Api.21/Controllers/ProductsController.cs?name=snippet_Get)]
 
 Without known conditions to safeguard against during action execution, returning a specific type could suffice. The preceding action accepts no parameters, so parameter constraints validation isn't needed.
 
-When known conditions need to be accounted for in an action, multiple return paths are introduced. In such a case, it's common to mix an [ActionResult](/dotnet/api/microsoft.aspnetcore.mvc.actionresult) return type with the primitive or complex return type. Either [IActionResult](#iactionresult-type) or [ActionResult\<T>](#actionresultt-type) are necessary to accommodate this type of action.
+When known conditions need to be accounted for in an action, multiple return paths are introduced. In such a case, it's common to mix an <xref:Microsoft.AspNetCore.Mvc.ActionResult> return type with the primitive or complex return type. Either [IActionResult](#iactionresult-type) or [ActionResult\<T>](#actionresultt-type) are necessary to accommodate this type of action.
+
+::: moniker range=">= aspnetcore-3.0"
+
+### Return IAsyncEnumerable\<T>
+
+In ASP.NET Core 2.2 or earlier versions, returning <xref:System.Collections.Generic.IAsyncEnumerable%601> from a web API action results in synchronous iteration by the serializer. The result is the blocking of calls and a potential for thread pool starvation. In ASP.NET Core 3.0 or later, returning `IAsyncEnumerable<T>` from a web API action is as efficient as returning <xref:System.Collections.Generic.IEnumerable%601>.
```

Comment:
Note that this behavior is based on the underlying value that's returned, not the declared typed of the method. I recommend that users declare the return type as `IAsyncEnumerable<>` where possible since that's the guarantee you want.